### DISCUSSION TITLE
collapse horizontal with dynamic width

### DISCUSSION THREAD

**Discussion by greenlevel** [twbs/bootstrap#37335] [Q&A]
Hi all,

I am using the horizontal collapse of bootstrap 5.2 but when reading the description i can only set a fixed width?

"Only the width on the child element is required."

I want to use a width in percentages. So it stretches all over the website depending on the screen size.
I tried to use 50% for example. but this doesn't seem to work.
Any idea how to solve this?
Thanks in advice!

**Comment by mdo** [comment]
I believe this is a limitation of CSS itself—it has to have a fixed width. You could manage that more with custom JS perhaps though.

**Reply by greenlevel** [reply]
Thanks for the quick feedback.

Good to know the limitations.  I will look into fixing it with JS.
